### PR TITLE
Fix shallow copy of typeutil.AppendFieldData

### PR DESCRIPTION
**Pull request by longjiquan**
Signed-off-by: dragondriver <jiquan.long@zilliz.com>
/kind improvement

### CODE REVIEW

**Comment by DragonDriver** [comment]
/run-checks
rerun ut

**Comment by DragonDriver** [comment]
rerun ut

File: internal/util/typeutil/schema.go [hunk 1]
```diff
@@ -275,20 +275,24 @@ func AppendFieldData(dst []*schemapb.FieldData, src []*schemapb.FieldData, idx i
 			switch srcVector := fieldType.Vectors.Data.(type) {
 			case *schemapb.VectorField_BinaryVector:
 				if dstVector.GetBinaryVector() == nil {
+					srcToCopy := srcVector.BinaryVector[idx*(dim/8) : (idx+1)*(dim/8)]
 					dstVector.Data = &schemapb.VectorField_BinaryVector{
-						BinaryVector: srcVector.BinaryVector[idx*(dim/8) : (idx+1)*(dim/8)],
+						BinaryVector: nil,
 					}
+					copy(dstVector.Data.(*schemapb.VectorField_BinaryVector).BinaryVector, srcToCopy)
```
Comment (congqixia) [inline internal/util/typeutil/schema.go:282]: `BinaryVector` is not pre-allocated, so copy will do nothing.  Maybe it shall be like this
```
BinaryVector: make([]byte, len(srcToCopy),
```

File: internal/util/typeutil/schema.go [hunk 2]
```diff
@@ -275,20 +275,24 @@ func AppendFieldData(dst []*schemapb.FieldData, src []*schemapb.FieldData, idx i
...
 					dstBinaryVector := dstVector.Data.(*schemapb.VectorField_BinaryVector)
 					dstBinaryVector.BinaryVector = append(dstBinaryVector.BinaryVector, srcVector.BinaryVector[idx*(dim/8):(idx+1)*(dim/8)]...)
 				}
 			case *schemapb.VectorField_FloatVector:
 				if dstVector.GetFloatVector() == nil {
+					srcToCopy := srcVector.FloatVector.Data[idx*dim : (idx+1)*dim]
 					dstVector.Data = &schemapb.VectorField_FloatVector{
 						FloatVector: &schemapb.FloatArray{
-							Data: srcVector.FloatVector.Data[idx*dim : (idx+1)*dim],
+							Data: nil,
```
Comment (congqixia) [inline internal/util/typeutil/schema.go:292]: Same issue here.

File: internal/util/typeutil/schema.go (hunk 1 above)
Comment (DragonDriver) [reply internal/util/typeutil/schema.go]: OK, got it. thx for pointing out this.

**Pull request merged**